packages/number-field/src/NumberField.ts

```diff
@@ -430,6 +430,17 @@ export class NumberField extends TextfieldBase {
         });
     }
 
+    private hasRecentlyReceivedPointerDown = false;
+
+    protected override handleInputElementPointerdown(): void {
+        this.hasRecentlyReceivedPointerDown = true;
+        this.updateComplete.then(() => {
```

Comment:
Won't this cause a delay in the event to complete?

Reply:
This is a reaction to the event having been triggered. Because it is in the promise (`updateComplete`) it will release the thread during the event and wait till the promise releases before reengaging the workflow.